ddl/index_cop.go

```diff
@@ -273,7 +273,14 @@ type copContext struct {
 	virtualColFieldTps  []*types.FieldType
 }
 
-func newCopContext(tblInfo *model.TableInfo, idxInfo *model.IndexInfo, sessCtx sessionctx.Context) (*copContext, error) {
+// FieldTypes is only used for test.
+// TODO(tangenta): refactor the operators to avoid using this method.
+func (c *copContext) FieldTypes() []*types.FieldType {
```

Comment:
Do you mean that operator can have filedTypes internally？

Reply:
Maybe make `copContext` public. I am not quite sure.